ql/src/java/org/apache/hadoop/hive/ql/metadata/SessionHiveMetaStoreClient.java

```diff
@@ -142,2416 +40,28 @@
  * so the readers of the objects in these maps should have the most recent view of the object.
  * But again, could be fragile.
  */
-public class SessionHiveMetaStoreClient extends HiveMetaStoreClientWithLocalCache implements IMetaStoreClient {
-  private static final Logger LOG = LoggerFactory.getLogger(SessionHiveMetaStoreClient.class);
-
+public class SessionHiveMetaStoreClient extends BaseMetaStoreClientProxy implements IMetaStoreClient {
```

Comment:
how about make the constructor private, and create a static method `newSessionHiveMetaStoreClient(Configuration conf, HiveMetaHookLoader hookLoader,
      Boolean allowEmbedded)`